src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +78,279 @@ public class AnalysisService : IDisposable
-        public string[] ActiveRules { get; set; }
-
-        /// <summary>
-        /// Gets or sets the path to a settings file (.psd1)
-        /// containing PSScriptAnalyzer settings.
-        /// </summary>
-        public string SettingsPath { get; set; }
+        private bool _hasInstantiatedAnalysisEngine;
 
-        #endregion
+        private PssaCmdletAnalysisEngine _analysisEngine;
 
-        #region Constructors
+        private CancellationTokenSource _diagnosticsCancellationTokenSource;
 
         /// <summary>
-        /// Construct a new AnalysisService object.
+        /// Construct a new AnalysisService.
         /// </summary>
-        /// <param name="analysisRunspacePool">
-        /// The runspace pool with PSScriptAnalyzer module loaded that will handle
-        /// analysis tasks.
-        /// </param>
-        /// <param name="pssaSettingsPath">
-        /// The path to the PSScriptAnalyzer settings file to handle analysis settings.
-        /// </param>
-        /// <param name="activeRules">An array of rules to be used for analysis.</param>
-        /// <param name="logger">Maintains logs for the analysis service.</param>
-        /// <param name="pssaModuleInfo">
-        /// Optional module info of the loaded PSScriptAnalyzer module. If not provided,
-        /// the analysis service will populate it, but it can be given here to save time.
-        /// </param>
-        private AnalysisService(
-            RunspacePool analysisRunspacePool,
-            string pssaSettingsPath,
-            IEnumerable<string> activeRules,
+        /// <param name="loggerFactory">Logger factory to create logger instances with.</param>
+        /// <param name="languageServer">The LSP language server for notifications.</param>
+        /// <param name="configurationService">The configuration service to query for configurations.</param>
+        /// <param name="workspaceService">The workspace service for file handling within a workspace.</param>
+        public AnalysisService(
+            ILoggerFactory loggerFactory,
             ILanguageServer languageServer,
             ConfigurationService configurationService,
-            ILogger logger,
-            PSModuleInfo pssaModuleInfo = null)
+            WorkspaceService workspaceService)
         {
-            _analysisRunspacePool = analysisRunspacePool;
-            SettingsPath = pssaSettingsPath;
-            ActiveRules = activeRules.ToArray();
+            _loggerFactory = loggerFactory;
+            _logger = loggerFactory.CreateLogger<AnalysisService>();
             _languageServer = languageServer;
             _configurationService = configurationService;
-            _logger = logger;
-            _pssaModuleInfo = pssaModuleInfo;
-            _mostRecentCorrectionsByFile = new ConcurrentDictionary<string, (SemaphoreSlim, Dictionary<string, MarkerCorrection>)>();
+            _workplaceService = workspaceService;
+            _analysisDelayMillis = 750;
```

Comment:
can this be a constant instead?

Reply:
I was wondering whether to put it in as a constructor parameter. It used to be inline where the task is. As a field, we could use reflection to change the setting even.